pbm/backup/oplog.go

```diff
@@ -23,24 +24,45 @@ func NewOplog(node *pbm.Node) *Oplog {
 	}
 }
 
-// SliceTo slice oplog starting from given timestamp
-func (ot *Oplog) SliceTo(ctx context.Context, w io.Writer, fromTs primitive.Timestamp) error {
+// SliceTo writes the oplog slice between given timestamps into the given w
+//
+// To be sure we have read ALL records up to the specified cluster time.
+// Specifically, to be sure that no operations from the past gonna came after we finished the slicing,
+// we have to tail until some record with ts > toTS. And it might be a noop.
+func (ot *Oplog) SliceTo(ctx context.Context, w io.Writer, from, to primitive.Timestamp) error {
 	clName, err := ot.collectionName()
 	if err != nil {
 		return errors.Wrap(err, "determine oplog collection name")
 	}
 	cl := ot.node.Session().Database("local").Collection(clName)
 
-	cur, err := cl.Find(ctx, bson.M{
-		"op": bson.M{"$ne": pbm.OperationNoop},
-		"ts": bson.M{"$gte": fromTs},
-	})
+	cur, err := cl.Find(ctx,
+		bson.M{
+			"ts": bson.M{"$gte": from},
+		},
+		options.Find().SetCursorType(options.Tailable),
+	)
```

Comment:
Shouldn't `"ts": bson.M{"$lte": to}` be in the query args here as well?
This would make it possible to remove the `primitive.CompareTimestamp(to, opts) > 0` check in the cursor iteration below too.

Reply:
Ah, I think that must be because the tailable cursor option is still on. I didn't notice it yesterday, so sorry.

As we're starting the query after all the target records are present, already, we don't have to tail any more. Just dump in one pass.

Reply:
I turned cursor intentionally. Because I'm not sure that we're starting the query after all the target records are present. We just start the query after the latest dump was finished. And there are no guarantees that no targeted records gonna came after we started the tailing (due network issues, node lag etc.). Ok, the probability of such behaviour not so high. But I'd prefer to be secure in that matter.